src/misc/lv_fs.c

```diff
@@ -144,11 +147,48 @@ lv_fs_res_t lv_fs_open(lv_fs_file_t * file_p, const char * path, lv_fs_mode_t mo
 
 void lv_fs_make_path_from_buffer(lv_fs_path_ex_t * path, char letter, const void * buf, uint32_t size)
 {
-    path->path[0] = letter;
-    path->path[1] = ':';
-    path->path[2] = 0;
-    path->buffer = buf;
-    path->size = size;
+    /*Make a path the contains both the address and the size.
+     *Also append .bin to make look like a real file path*/
+#ifdef LV_ARCH_64
+    lv_snprintf(path->path, sizeof(path->path), "%c:%" LV_PRIu64 "-%" LV_PRIu32 ".bin", letter,
+                (uint64_t)((lv_uintptr_t) buf), size);
+#else
+    lv_snprintf(path->path, sizeof(path->path), "%c:%" LV_PRIu32 "-%" LV_PRIu32 ".bin", letter,
+                (uint32_t)((lv_uintptr_t) buf), size);
+#endif
```

Comment:
```suggestion
    lv_snprintf(path->path, sizeof(path->path), "%c:%zu-%" LV_PRIu32 ".bin", letter,
                (size_t)buf, size);
```

Easier to read, what do you think ?

Reply:
Let's see if it passes all the CI 🙃

Reply:
Mmm, on 64-bit this is passing the wrong size type for the format specifier. `size_t` is 64 bit on 64 bit arch. The format specifier is 32 bit. I think this is not a good idea.

Reply:
Ignore my above comment. I was looking at the wrong format specifier. It's a good improvement.